test/conformance/testing/uur/utils.h

```diff
@@ -42,6 +48,39 @@ inline std::string GetPlatformAndDeviceName(ur_device_handle_t device) {
     return GetPlatformName(GetPlatform()) + "__" + GetDeviceName(device);
 }
 
+template <class T, class ObjectTy, class InfoTy, class Callable>
+std::optional<T> GetInfo(ObjectTy object, InfoTy info, Callable cb) {
+    size_t infoSize = 0;
+    if (cb(object, info, 0, nullptr, &infoSize)) {
+        return std::nullopt;
+    }
+    if (infoSize == 0 || infoSize != sizeof(T)) {
+        return std::nullopt;
+    }
+    T queryValue{};
+    if (cb(object, info, sizeof(T), &queryValue, nullptr)) {
+        return std::nullopt;
+    }
+    return queryValue;
+}
+
+template <class T>
+std::optional<uint32_t> GetObjectReferenceCount(T object) {
+    if constexpr (std::is_same_v<T, ur_context_handle_t>) {
+        return GetInfo<uint32_t>(object, UR_CONTEXT_INFO_REFERENCE_COUNT, urContextGetInfo);
+    }
+    if constexpr (std::is_same_v<T, ur_device_handle_t>) {
+        return GetInfo<uint32_t>(object, UR_DEVICE_INFO_REFERENCE_COUNT, urDeviceGetInfo);
+    }
+    if constexpr (std::is_same_v<T, ur_event_handle_t>) {
+        return GetInfo<uint32_t>(object, UR_EVENT_INFO_REFERENCE_COUNT, urEventGetInfo);
+    }
+    if constexpr (std::is_same_v<T, ur_queue_handle_t>) {
+        return GetInfo<uint32_t>(object, UR_QUEUE_INFO_REFERENCE_COUNT, urQueueGetInfo);
+    }
```

Comment:
Should we also handle the other handle types so it doesn't need modified later?

Reply:
Done